### PR TITLE
[BEAM-7093] Support Spark 3 in Spark runner

### PR DESCRIPTION
**Pull request by iemejia**
Let's get the party started
R: @ibzib

### CODE REVIEW

**Comment by iemejia** [comment]
Run Seed Job

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark StructuredStreaming ValidatesRunner

**Comment by iemejia** [comment]
Run Spark Runner Nexmark Tests

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Streaming

**Comment by iemejia** [comment]
Run Python Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Go Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark ValidatesRunner Java 11

**Comment by iemejia** [comment]
Run Seed Job

**Comment by iemejia** [comment]
Run Spark ValidatesRunner Java 11

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark StructuredStreaming ValidatesRunner

**Comment by iemejia** [comment]
Run Spark Runner Nexmark Tests

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Streaming

**Comment by iemejia** [comment]
Run Python Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Go Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Seed Job

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark StructuredStreaming ValidatesRunner

**Comment by iemejia** [comment]
Run Spark Runner Nexmark Tests

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Streaming

**Comment by iemejia** [comment]
Run Python Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Go Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Batch

**Comment by ibzib** [comment]
run seed job

**Comment by ibzib** [comment]
Run Java Spark PortableValidatesRunner Batch

File: runners/spark/spark_runner.gradle [hunk 1]
```diff
@@ -90,24 +163,25 @@ dependencies {
   compile project(":sdks:java:fn-execution")
   compile library.java.vendored_grpc_1_26_0
   compile library.java.vendored_guava_26_0_jre
-  provided library.java.spark_core
```
Comment (ibzib) [inline runners/spark/spark_runner.gradle:93]: We should remove these from BeamModulePlugin if they're no longer used.

**Review by ibzib — APPROVED** [review]
Looks like you've addressed most of my comments from the previous iteration of this PR. Some files (including the release scripts) are missing here, but it's fine to put those in a separate PR.

LGTM, now let's see if we can get the tests to pass.

**Comment by ibzib** [comment]
@iemejia any updates on this?

**Comment by iemejia** [comment]
Yes, the recently merged spark history server PR broke backwards compatibility. I am preparing a fix for that and afterwards I expect to rebase this one.

**Comment by ibzib** [comment]
> Yes, the recently merged spark history server PR broke backwards compatibility. I am preparing a fix for that and afterwards I expect to rebase this one.

Thanks Ismaël. I'll be more careful merging future Spark PRs so they don't break this again..

**Comment by iemejia** [comment]
Don't worry Kyle it is easy to miss these kind of compatibility issues without automated tests, but that's coming soon. For info #14132 is out with the backwards compatibility fix.

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark StructuredStreaming ValidatesRunner

**Comment by iemejia** [comment]
Run Spark Runner Nexmark Tests

**Comment by iemejia** [comment]
Run Java Spark PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Python Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Go Spark ValidatesRunner

**Comment by iemejia** [comment]
The failing VR `WindowTest.testMergingCustomWindowsWithoutCustomWindowTypes` is being tracked by [BEAM-11887](https://issues.apache.org/jira/browse/BEAM-11887)

**Comment by iemejia** [comment]
Run Python_PVR_Flink PreCommit

**Comment by iemejia** [comment]
Run Seed Job

**Comment by iemejia** [comment]
Run GoPortable PreCommit

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark StructuredStreaming ValidatesRunner

**Pull request closed without merging**